src/utilities/sql/SqlFile_Impl.cpp

```diff
@@ -378,31 +334,17 @@ namespace openstudio{
 
     }
 
-    void SqlFile_Impl::execAndThrowOnError(const std::string &t_stmt)
-    {
-      char *err = nullptr;
-      if (sqlite3_exec(m_db, t_stmt.c_str(), nullptr, nullptr, &err) != SQLITE_OK)
-      {
-        std::string errstr;
-
-        if (err)
-        {
-          errstr = err;
-          sqlite3_free(err);
-        }
-
-        throw std::runtime_error("Error executing SQL statement: " + t_stmt + " " + errstr);
-      }
-    }
-
     bool SqlFile_Impl::connectionOpen() const
     {
       return m_connectionOpen;
     }
 
     int SqlFile_Impl::getNextIndex(const std::string &t_tableName, const std::string &t_columnName)
     {
-      boost::optional<int> maxindex = execAndReturnFirstInt("select max(" + t_columnName + ") from " + t_tableName);
+      // Interestingly, you CANNOT bind any database identifier (such as the table name / column name) but only litteral values...
+      // boost::optional<int> maxindex = execAndReturnFirstInt("SELECT MAX( ? ) FROM ?", t_columnName, t_tableName);
+      std::string query = "SELECT MAX(" + t_columnName + ") FROM " + t_tableName + ";";
+      boost::optional<int> maxindex = execAndReturnFirstInt(query);
```

Comment:
One where I couldn't use parameters, so no escaping, but it's actually safe here, since table/column names are E+ defined and safe